docs: Add and refine developer contribution guidelines (#1356)
omegat-org/omegat: src_docs/developer/03.GitWorkFlows.md

# Git Workflow (for New Contributors)

> If you're planning to contribute to OmegaT — whether it's code, documentation, or translation — we’re happy to have you on board!

Reviewer: It would be nicer to always use the same type of apostrophes.
```suggestion
If you're planning to contribute to OmegaT — whether it's code, documentation, or translation — we're happy to have you on board!
```

Reviewer: There are a few other instances of `’` but we seem to use mostly `'`, maybe we can do a quick search and replace in the edited dev docs?

```

4. **Create your topic branch:**

```bash
   git checkout -b topic/your-username/area/short-description
```   

---

## ✅ Before Opening a Pull Request

### 🔍 Check What You Changed

Use our custom Gradle task to check which files you've modified.

**In your IDE:**

- Open the Gradle Tasks panel
- Expand `omegat workflow`
- Run `changeOnBranch`

**From the terminal:**

- Run `./gradlew changeOnBranch`

---

### 🎨 Format Your Code

Use Spotless to automatically format only your changes.

**In your IDE:**

- Open the Gradle Tasks panel
- Expand `omegat workflow`
- Run `spotlessChangedApply`

**From the terminal:**

- Run `./gradlew spotlessChangedApply`

This ensures your code follows the project's style and avoids noisy diffs.

---

## 📬 Submit Your Pull Request

1. **Push your topic branch to GitHub:**

```bash
   git push origin topic/your-username/area/short-description
```

2. **Open a Pull Request:**

  - Visit your fork on GitHub
  - Click "Compare & pull request"
  - Add a clear title and explanation
  - Link to any related issues or discussions

Tip: You can open a **draft pull request** if you're still working on it but want feedback early.

---

## 💬 Need Help?

> Feel free to ask on our developers’ mailing list:  
> 👉 http://lists.sourceforge.net/lists/listinfo/omegat-development
> 
> We’re happy to support you!

Reviewer: ```suggestion
Feel free to ask on the developers mailing list:  
👉 http://lists.sourceforge.net/lists/listinfo/omegat-development

We're happy to support you!
```
